clustered/integration-test/src/test/java/org/ehcache/clustered/management/ClusteringManagementServiceTest.java

```diff
@@ -190,26 +190,43 @@ public void test_stats_collection() throws Exception {
     cache1.get("key1");
     cache1.get("key2");
 
-    // get the stats (we are getting the primitive counter, not the sample history)
-    ContextualStatistics[] stats = waitForNextStats();
-    Sample<Long>[] samples = stats[0].getStatistic(CounterHistory.class, "Cache:HitCount").getValue();
 
-    assertThat(stats.length, equalTo(1));
-    assertThat(stats[0].getContext().get("cacheName"), equalTo("cache-1"));
-    assertThat(samples[0].getValue(), equalTo(2L));
+    long val = 0;
+
+    // it could be several seconds before the sampled stats could become available
+    // let's try until we find the correct value : 2
+    do {
+
+      // get the stats (we are getting the primitive counter, not the sample history)
+      ContextualStatistics[] stats = waitForNextStats();
+      Sample<Long>[] samples = stats[0].getStatistic(CounterHistory.class, "Cache:HitCount").getValue();
+
+      if(stats.length == 1 && stats[0].getContext().get("cacheName").equals("cache-1")) {
+        val = samples[0].getValue();
+      }
+    } while(val != 2);
```

Comment:
Can you add a timeout to the test then? Don't want to see it loop forever the day it is effectively broken by an invalid change.